AX: Update accname for presentational images to be ignored despite fallbacks

rahimabdi opened the pull request:
#### fbcb00b98357a19fda3018f071dd62ebeac76c57
<pre>
AX: Update accname for presentational images to be ignored despite fallbacks
<a href="https://bugs.webkit.org/show_bug.cgi?id=289290">https://bugs.webkit.org/show_bug.cgi?id=289290</a>
<a href="https://rdar.apple.com/146429365">rdar://146429365</a>

Reviewed by Tyler Wilcock.

Where an image element&apos;s alt attribute is set to the empty string (i.e. alt=&quot;&quot;), it is intended that the image
is decorative and not conveyed to assistive technology users. Currently, for images
with empty alt that have additional labeling techniques (such as title or aria-label), the image is not ignored
and the additional attributes for labeling are honored.

This commit ensures that decorative images supplied with empty alt are ignored despite the presence
of other labeling methods such as the title attribute.

* LayoutTests/accessibility/img-with-empty-alt-is-ignored-without-fallbacks.html: Added.
* LayoutTests/accessibility/img-no-alt-not-ignored-with-title.html: Deleted.
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::computeIsIgnored const):

Canonical link: <a href="https://commits.webkit.org/292105@main">https://commits.webkit.org/292105@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d9daeaaae40d073d9b21fd7dd156aa134e4eca1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94884 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14474 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4321 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99901 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45372 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96931 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14757 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22895 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72373 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29665 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97885 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10996 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85668 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52748 "Found 1 new API test failure: /TestWebCore:GStreamerTest.capsFromCodecString (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10700 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3387 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44713 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80958 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3482 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101942 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21916 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16002 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81412 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22163 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81693 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80760 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20194 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25322 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2719 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/15148 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21891 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27009 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21551 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25022 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23291 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

twilco reviewed (approved):
r+ with bots passing

rahimabdi committed:
AX: Update accname for presentational images to be ignored despite fallbacks
https://bugs.webkit.org/show_bug.cgi?id=289290
rdar://146429365

Reviewed by Tyler Wilcock.

Where an image element's alt attribute is set to the empty string (i.e. alt=""), it is intended that the image
is decorative and not conveyed to assistive technology users. Currently, for images
with empty alt that have additional labeling techniques (such as title or aria-label), the image is not ignored
and the additional attributes for labeling are honored.

This commit ensures that decorative images supplied with empty alt are ignored despite the presence
of other labeling methods such as the title attribute.

* LayoutTests/accessibility/img-with-empty-alt-is-ignored-without-fallbacks.html: Added.
* LayoutTests/accessibility/img-no-alt-not-ignored-with-title.html: Deleted.
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::computeIsIgnored const):

Canonical link: https://commits.webkit.org/292105@main
